internal/cli/models.go

```diff
@@ -489,6 +491,30 @@ type (
 	}
 )
 
+// getKnownFields uses reflection to extract JSON field names from BlueprintProperty struct
+func getKnownFields(bp *BlueprintProperty) map[string]bool {
+	knownFields := make(map[string]bool)
+	t := reflect.TypeOf(*bp)
+	
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		
+		// Get the JSON tag
+		jsonTag := field.Tag.Get("json")
+		if jsonTag == "" || jsonTag == "-" {
+			continue // Skip fields without JSON tags or with "-"
+		}
+		
+		// Handle "fieldname,omitempty" format
+		fieldName := strings.Split(jsonTag, ",")[0]
```

Comment:
Maybe use `strings.Cut` instead

```suggestion
		fieldName, _, _ = strings.Cut(s, ",")
```